docs/zh/FAQ/部署&开发遇到的常见问题.md

```diff
@@ -0,0 +1,319 @@
+
+### 1. windows怎么执行build.sh?
+安装[Git Bash](https://git-for-windows.github.io/)，然后运行 “./build.sh” 注意前面 “./”
+
+### 2. 本地运行时Portal一直报Env is down. 
+
+默认config service启动在8080端口，admin service启动在8090端口。请确认这两个端口是否被其它应用程序占用。
+
+如果还伴有异常信息：org.springframework.web.client.HttpClientErrorException: 405 Method Not Allowed，一般是由于本地启动了`ShadowSocks`，因为`ShadowSocks`默认会占用8090端口。
+
+1.1.0版本增加了**系统信息**页面，可以通过`管理员工具` -> `系统信息`查看当前各个环境的Meta Server以及admin service信息，有助于排查问题。
+
+### 3. admin server 或者 config server 注册了内网IP，导致portal或者client访问不了admin server或config server
+
+分布式部署的时候，`apollo-configservice`和`apollo-adminservice`需要把自己的IP和端口注册到Meta Server（apollo-configservice本身）。
+
+Apollo客户端和Portal会从Meta Server获取服务的地址（IP+端口），然后通过服务地址直接访问。
+
+所以如果实际部署的机器有多块网卡（如docker），或者存在某些网卡的IP是Apollo客户端和Portal无法访问的（如网络安全限制），那么我们就需要在`apollo-configservice`和`apollo-adminservice`中做相关限制以避免Eureka将这些网卡的IP注册到Meta Server。
+
+#### 方案一： 忽略某些网卡
+具体文档可以参考[Ignore Network Interfaces](http://projects.spring.io/spring-cloud/spring-cloud.html#ignore-network-interfaces)章节。具体而言，就是分别编辑[apollo-configservice/src/main/resources/application.yml](https://github.com/ctripcorp/apollo/blob/master/apollo-configservice/src/main/resources/application.yml)和[apollo-adminservice/src/main/resources/application.yml](https://github.com/ctripcorp/apollo/blob/master/apollo-adminservice/src/main/resources/application.yml)，然后把需要忽略的网卡加进去。
+
+如下面这个例子就是对于`apollo-configservice`，把docker0和veth.*的网卡在注册到Eureka时忽略掉。
+
+```yaml
+    spring:
+      application:
+          name: apollo-configservice
+      profiles:
+        active: ${apollo_profile}
+      cloud:
+        inetutils:
+          ignoredInterfaces:
+            - docker0
+            - veth.*
+```
+> 注意，对于application.yml修改时要小心，千万不要把其它信息改错了，如spring.application.name等。
+
+#### 方案二：强制指定admin server和config server向eureka注册的IP
+
+可以修改startup.sh，通过JVM System Property在运行时传入，如`-Deureka.instance.ip-address=${指定的IP}`，或者也可以修改apollo-adminservice或apollo-configservice 的bootstrap.yml文件，加入以下配置
+
+``` yaml
+eureka:
+  instance:
+    ip-address: ${指定的IP}
+```
+#### 方案三：强制指定admin server和config server向eureka注册的IP和Port
+
+可以修改startup.sh，通过JVM System Property在运行时传入，如`-Deureka.instance.homePageUrl=http://${指定的IP}:${指定的Port}`，或者也可以修改apollo-adminservice或apollo-configservice 的bootstrap.yml文件，加入以下配置
+
+``` yaml
+eureka:
+  instance:
+    homePageUrl: http://${指定的IP}:${指定的Port}
+    preferIpAddress: false
+```
+
+做完上述修改并重启后，可以查看Eureka页面（http://${config-service-url:port}）检查注册上来的IP信息是否正确。
+
+> 注：如果Apollo部署在公有云上，本地开发环境无法连接，但又需要做开发测试的话，客户端可以升级到0.11.0版本及以上，然后通过`-Dapollo.configService=http://config-service的公网IP:端口`来跳过meta service的服务发现，记得要对公网 SLB 设置 IP 白名单，避免数据泄露
+
+
+### 4. Portal如何增加环境？
+
+#### 4.1 1.6.0及以上的版本
+
+1.6.0版本增加了自定义环境的功能，可以在不修改代码的情况增加环境
+
+1. protaldb增加环境，参考[分布式部署指南 2.1.3.1一节](https://github.com/ctripcorp/apollo/wiki/%E5%88%86%E5%B8%83%E5%BC%8F%E9%83%A8%E7%BD%B2%E6%8C%87%E5%8D%97#2131-%E8%B0%83%E6%95%B4apolloportaldb%E9%85%8D%E7%BD%AE)
```

Comment:
I think we also need to update the internal links?

Reply:
Have changed almost all internal links.

Reply:
It looks like the internal links doesn't work when viewed in github repository, is it possible to make it work both in github repository and in the standalone website?

Reply:
[docsify](https://docsify.js.org/#/) seems not support that. When a link work well in docsify, it cannot work in github repository because of route is not matched.